### PR TITLE
Finalized nav bar

### LINKED ISSUE #591: Finalize navbar UI and use it in both React and CMS-based pages
[Design (at top)](https://projects.invisionapp.com/d/main#/console/19928472/417192796/preview)

In addition to matching the navbar UI to the designs, we will want to create a separate bundle for the navbar and render it on React- and CMS-based pages to avoid creating 2 parallel implementations. We did this in xPro (PR: https://github.com/mitodl/mitxpro/pull/130).

#### Acceptance criteria
- [ ] Navbar UI matches the designs
- [ ] "Sign In"/"Create Account" button styling updated to match other buttons in the designs (rounded corners, color background instead instead of grey)
- [ ] Buttons have expected results upon clicking:
  - [ ] "Apply Now" - ~~opens a drawer with the new application form (https://github.com/mitodl/mitxpro/pull/524)~~ This will simply redirect the user to the application dashboard (`/applications`)
  - [ ] (pending decision) "Bootcamps" - navigates to the page configured as the CMS home page
  - [ ] (pending decision) "How to Apply" - ???
  - [ ] "Sign In"/"Create Account" - obvious
- [ ] Navbar separated into separate bundle and rendered on React- and CMS-based pages
- [ ] Existing navbar code in HTML templates removed

### PR DESCRIPTION
**Pull request by gsidebo**
#### What are the relevant tickets?
Closes #591 

#### What's this PR do?
- Fixes the navbar UI to more closely match the designs + design feedback
- Renders the same navbar in the React- and Django-based apps
- Fixes the positioning of notifications

#### How should this be manually tested?
- Look at the navbar in multiple screen widths, logged in and logged out. Check it against the designs
- Add a site-wide notification in the CMS. Make sure it shows up directly below the navbar

#### Any background context you want to provide?
- The "Sign In"/"Create Account"/"Sign Out" buttons are styled differently from the designs. This was based on some verbal feedback
- The "How to Apply" link will not work unless you add a resource page in the CMS entitled "Apply"
- This PR also fixes a mobile-width issue with the "View Statements" link
- ~~We might want to use an expandable menu in the mobile view like we do in xpro. That can be done in this PR or a future one~~ ~~This expandable menu was actually in an Invision design that I didn't see. Implementing that now~~ Done

#### Screenshots (if appropriate)
ss 2020-06-18 at 14 28 53
ss 2020-06-18 at 14 29 03
ss 2020-06-18 at 14 31 20
ss 2020-06-18 at 14 31 32
ss 2020-06-18 at 15 03 32
ss 2020-06-18 at 15 03 14
ss 2020-06-18 at 14 33 18

### CODE REVIEW

**Comment by gsidebo** [comment]
@abdulkdawson Please review the screenshots ^^^. I figure we're probably going to want to do something with the 2-row mobile view

**Comment by gsidebo** [comment]
Updated right side button screenshots: 

ss 2020-06-18 at 10 02 25
ss 2020-06-18 at 10 02 38

**Comment by gsidebo** [comment]
@abdulkdawson @Ferdi Screenshots updated. Please review ^^^

**Comment by noisecapella** [comment]
Not sure if this was some of the verbal feedback but it looks like the "Apply Now" button is bold in the mockup but not bold in the PR

**Comment by noisecapella** [comment]
Functionality looks great. I'll dive into the code/tests tomorrow

File: static/scss/_inputs.scss [hunk 1]
```diff
@@ -28,16 +28,68 @@
...
+  }
+}
+
+a.btn-styled {
+  display: inline-block;
+  color: $white;
+  text-decoration: none;
+
+  &:focus {
+    //outline-color: inherit;
```
Comment (noisecapella) [inline static/scss/_inputs.scss:56]: Stray comment

File: main/utils.py [hunk 2]
```diff
@@ -261,6 +261,11 @@ def get_field_names(model):
     ]
 
 
+def filter_dict_by_key_set(dict_to_filter, key_set):
```
Comment (noisecapella) [inline main/utils.py:264]: Can we add a test for this?

Comment (gsidebo) [reply main/utils.py:264]: LOL, accidentally added this from another branch. Test added anyway

**Pull request merged**